static/js/popovers.js

```diff
@@ -547,6 +547,14 @@ exports.register_click_handlers = function () {
         e.stopPropagation();
     });
 
+    $('body').on('click', '.mark_stream_as_read', function (e) {
+        var stream = $(e.currentTarget).parents('ul').attr('data-name');
+        //var topic = $(e.currentTarget).attr('data-topic-name');
```

Comment:
Yeah, I was planning on doing so; All I really need for it is to know how topics associated with a stream are stored (i.e. when I get the stream object in `update_message_flags` if I can directly access it then it would be very easy)

I guess more generally I'm still not 100% clear on where things like the different properties of Stream.object and such are defined/documented; most of what I've written so far has just been using existing code along with trial & error.